### PR TITLE
Authentication rules

### PR DESCRIPTION
**Pull request by AnthoBalitrand**
## Description

Added policies.AuthenticationRule class to be able to retrieve / update authentication rules. 

## Motivation and Context

This rulebase was not available yet on the tool. 

## How Has This Been Tested?

Tested on production PaloAlto / Panorama deployment. 

## Screenshots (if appropriate)

N/A

## Types of changes

- New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes if appropriate.
- [ ] All new and existing tests passed.
--> Lots of warnings running existing tests (on other non-modified parts of the code)

### CODE REVIEW

File: panos/policies.py [hunk 1]
```diff
@@ -44,6 +44,7 @@ class Rulebase(VersionedPanObject):
         "policies.PolicyBasedForwarding",
         "policies.SecurityRule",
         "policies.DecryptionRule",
+        "policies.AuthenticationRule",
```
Comment (shinmog) [inline panos/policies.py:47]: Please add this line to PreRulebase and PostRulebase also.

**Review by shinmog — CHANGES_REQUESTED** [review]
This looks pretty good, but there's a lot of checks I don't want to do manually, so I'm really hoping that when you correct the few things I've mentioned, it will run the unit tests and save me a quite a bit of work...

File: panos/policies.py [hunk 2]
```diff
@@ -947,6 +948,118 @@ def _setup_opstate(self):
...
+
+        params.append(VersionedParamPath("description", path="description"))
+        params.append(VersionedParamPath("uuid", exclude=True))
+        params[-1].add_profile("9.0.0", vartype="attrib", path="uuid")
+        params.append(
+            VersionedParamPath("source_zones", vartype="member", path="from",)
+        )
+        params.append(
+            VersionedParamPath(
+                "source", default=["any", ], vartype="member", path="source"
```
[the remark this answers is not in the document]
Comment (AnthoBalitrand) [reply panos/policies.py]: Ok, tests run, param name + docstring fixed by commit #3e129ae

File: panos/policies.py (hunk 1 above)
Comment (AnthoBalitrand) [reply panos/policies.py:47]: Thanks for your feedback. I don't understand why ?

Comment (shinmog) [reply panos/policies.py:47]: `policies.Rulebase` is for NGFW rules.  But Panorama has `policies.PreRulebase` and `policies.PostRulebase`.  So if it's not there in those two, then the rule is not useable / accessible for Panorama.

**Comment by shinmog** [comment]
Good the tests ran this time, and passed.  Just need to fix the formatting issue using `black .` and this can be merged.

File: panos/policies.py (hunk 1 above)
Comment (AnthoBalitrand) [reply panos/policies.py:47]: Yes but `policies.PreRulebase` and `policies.PostRulebase` inherits from `policies.Rulebase` (just adding an _xpaths profile. 
It works actually pretty well with Panorama !

**Comment by shinmog** [comment]
@AnthoBalitrand 

Ok, so attempting to run the code formatter didn't quite work.  It changed more than just what you added, which means it might not have been using this repo's black config file..?  But now there's also a conflict from some of the other PRs I've been working through today.

**Comment by AnthoBalitrand** [comment]
I'll close this one and open another pull request.

**Pull request closed without merging**